src/os_linux.cpp

```diff
@@ -98,6 +98,10 @@ static SigAction installed_sigaction[64];
 
 const size_t OS::page_size = sysconf(_SC_PAGESIZE);
 const size_t OS::page_mask = OS::page_size - 1;
+const int OS::prot_read = PROT_READ;
+const int OS::prot_write = PROT_WRITE;
+const int OS::prot_copy = PROT_NONE;
```

Comment:
Seems like a leaky abstraction. Caller does not need to know about prot_copy, especially on Linux.
I suggest not introducing new constants and keep using generic PROT_{READ,WRITE,EXEC}, but in macOS implementation, convert them to proper `vm_protect` flags.